core/trino-main/src/test/java/io/trino/sql/planner/AbstractPredicatePushdownTest.java

```diff
@@ -65,14 +65,18 @@ public void testNonStraddlingJoinExpression()
         assertPlan(
                 "SELECT * FROM orders JOIN lineitem ON orders.orderkey = lineitem.orderkey AND cast(lineitem.linenumber AS varchar) = '2'",
                 anyTree(
-                        join(INNER, ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
-                                anyTree(
-                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
-                                anyTree(
-                                        filter("cast(LINEITEM_LINENUMBER as varchar) = VARCHAR '2'",
-                                                tableScan("lineitem", ImmutableMap.of(
-                                                        "LINEITEM_OK", "orderkey",
-                                                        "LINEITEM_LINENUMBER", "linenumber")))))));
+                        innerJoin()
+                                .equiCriteria("ORDERS_OK", "LINEITEM_OK")
+                                .left(
+                                        anyTree(
+                                                tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))))
+                                .right(
+                                        anyTree(
+                                                filter("cast(LINEITEM_LINENUMBER as varchar) = VARCHAR '2'",
+                                                        tableScan("lineitem", ImmutableMap.of(
+                                                                "LINEITEM_OK", "orderkey",
+                                                                "LINEITEM_LINENUMBER", "linenumber")))))
+                                .build()));
```

Comment:
nit: we probably could get rid of `build` if `anyTree` and others accepted interface like `PatternBuilder` or `Provider<PlanMatchPattern>`.
Not needed now

Reply:
Yes, this is one of possibility to remove `build` call and this kind of change could be done not only for `anyTree`.